plugins/fclrave/fclspace.cpp

```diff
@@ -392,6 +392,34 @@ void FCLSpace::SynchronizeWithAttached(const KinBody &body)
     }
 }
 
+void FCLSpace::SynchronizeExcluded(const KinBodyConstPtr& pbodyexcluded)
+{
+    // We synchronize only the initialized bodies, which differs from oderave
+    for (const KinBodyConstPtr& pbody : _vecInitializedBodies) {
+        if (!pbody) {
+            continue;
+        }
+        if (pbody == pbodyexcluded) {
+            continue;
+        }
+        Synchronize(*pbody);
+    }
+}
+
+void FCLSpace::SynchronizeLink(const KinBody::Link &link)
+{
+    const KinBody &body = *link.GetParent();
+    FCLKinBodyInfoPtr& pinfo = GetInfo(body);
+    if( !pinfo ) {
+        return;
+    }
+    if( pinfo->nLastStamp != body.GetUpdateStamp() ) {
```

Comment:
What is this check for? If `FCLKinBodyInfo.nLastStamp` is equal to body's update stamp, does it imply its member's `linkInfo.nLastStam` is also the same?

`linkInfo.nLastStamp != body.GetUpdateStamp()` is checked inside `_SynchronizeLink` to decide actual synchronization computation, we want to skip it if `pinfo->nLastStamp` equals `body.GetUpdateStamp()` in addition to `linkInfo.nLastStamp` condition?

Reply:
`linkInfo.nLastStamp` is the same as or newer than `FCLKinBodyInfo.nLastStamp`.

It was to early skip in the case where there are no link changes at all, but the effect is too small compared with that in `FCLSpace::_Synchronize`.
So, I removed the check to not cause confusion.